atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/SweeperServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+        TableReference tableRef = TableReference.createFromFullyQualifiedName(tableName);
+
+        Preconditions.checkArgument(TableReference.isFullyQualifiedName(tableName),
+                String.format("Table name %s is not fully qualified", tableName));
+        Preconditions.checkState(specificTableSweeper.getKvs().getAllTableNames().contains(tableRef),
+                String.format("Table requested to sweep %s does not exist", tableName));
+
+        return specificTableSweeper.runOnceForTable(new TableToSweep(tableRef, null), false, Optional.empty());
+    }
+
+    @Override
+    public boolean sweepTableFromStartRow(String tableName, String startRow) {
+        TableReference tableRef = TableReference.createFromFullyQualifiedName(tableName);
+
+        Preconditions.checkArgument(TableReference.isFullyQualifiedName(tableName),
+                "Table name is not fully qualified");
+        Preconditions.checkState(specificTableSweeper.getKvs().getAllTableNames().contains(tableRef),
+                "Table requested to sweep does not exist");
+
+        ImmutableSweepProgress sweepProgress = ImmutableSweepProgress.builder().tableRef(tableRef)
+                .staleValuesDeleted(0)
+                .cellTsPairsExamined(0)
+                .minimumSweptTimestamp(0)
+                .startRow(decodeStartRow(startRow))
+                .build();
+
+        return specificTableSweeper.runOnceForTable(new TableToSweep(tableRef, sweepProgress), false, Optional.empty());
+    }
+
+    @Override
+    public boolean sweepTableFromStartRowWithBatchConfig(String tableName,
+            String startRow,
+            @Nullable int maxCellTsPairsToExamine,
+            @Nullable int candidateBatchSize,
+            @Nullable int deleteBatchSize) {
+        TableReference tableRef = TableReference.createFromFullyQualifiedName(tableName);
+
+        Preconditions.checkArgument(TableReference.isFullyQualifiedName(tableName),
+                "Table name is not fully qualified");
+        Preconditions.checkState(specificTableSweeper.getKvs().getAllTableNames().contains(tableRef),
+                "Table requested to sweep does not exist");
+
+        ImmutableSweepBatchConfig sweepBatchConfig = ImmutableSweepBatchConfig.builder()
+                .maxCellTsPairsToExamine(maxCellTsPairsToExamine)
+                .candidateBatchSize(candidateBatchSize)
+                .deleteBatchSize(deleteBatchSize).build();
+
+        ImmutableSweepProgress sweepProgress = ImmutableSweepProgress.builder().tableRef(tableRef)
+                .staleValuesDeleted(0)
+                .cellTsPairsExamined(0)
+                .minimumSweptTimestamp(0)
+                .startRow(decodeStartRow(startRow))
+                .build();
+
+        return specificTableSweeper.runOnceForTable(
+                new TableToSweep(tableRef, sweepProgress),
+                false,
+                Optional.of(sweepBatchConfig));
+    }
```

Comment:
Hmm. We're using false for `saveSweepResults()`, but the original sweep CLI does actually update the priority table (`SweepCommand:193`) - would like to verify we're intending the semantics to be a bit different!

@gbonik for SA / thoughts